modules/local_storage/commerce_stock_local.module

```diff
@@ -40,14 +43,17 @@ function _commerce_stock_local_update_stock_level_queue() {
   $llq_batchsize = \Drupal::state()->get('commerce_stock_local.llq_batchsize');
   $llq_batchsize = !empty($llq_batchsize) ? $llq_batchsize : 50;
 
-  $result = \Drupal::entityQuery('commerce_product_variation')
+  /** @var \Drupal\Core\Entity\EntityStorageInterface $storage */
+  $storage = \Drupal::entityTypeManager()->getStorage('commerce_product_variation');
+
+  $result = $storage->getQuery()
     ->condition('variation_id', $level_last_id, '>')
     ->condition('status', 1, '=')
     ->sort('variation_id', 'ASC')
     ->range(0, $llq_batchsize)
     ->execute();
   foreach ($result as $pid) {
-    $queue->createItem($pid);
+    $queue->createItem($storage->load($pid));
```

Comment:
I am not sure how the $queue handles this but if it stores the fully loaded entity this is going to be much more expensive then storing entity type "commerce_product_variation" + $pid. Also we need to think about what to do with other purchasable entities. Lets get this in as is and talk it through next time we catchup.